client/src/components/Form/Elements/FormText.vue

```diff
@@ -0,0 +1,166 @@
+<template>
+    <div>
+        <b-alert class="mt-2" v-if="errorMessage" :show="dismissCountDown" variant="info" @dismissed="resetAlert">
+            {{ errorMessage }}
+        </b-alert>
+        <b-row align-v="center">
+            <b-col>
+                <component
+                    :is="componentName"
+                    :id="id"
+                    :readonly="readonly"
+                    :placeholder="placeholder"
+                    :style="style"
+                    :type="type"
+                    v-model="currentValue"
+                    @change="onInputChange" />
+
+                <datalist v-if="datalist && !area && !multiple" :id="`${id}-datalist`">
+                    <option v-for="d in datalist" value="d"></option>
+                </datalist>
+            </b-col>
+        </b-row>
+    </div>
+</template>
+
+<script>
+import Utils from "utils/utils";
+import Ui from "mvc/ui/ui-misc";
+
+export default {
+    props: {
+        value: {
+            // String; Array for multiple
+            required: false,
+            default: "",
+        },
+        id: {
+            type: String,
+            required: false,
+        },
+        type: {
+            type: String,
+            required: false,
+            default: "text",
+        },
+        area: {
+            // <textarea> instead of <input> element
+            type: Boolean,
+            required: false,
+            default: false,
+        },
+        multiple: {
+            // Allow multiple entries to be created
+            type: Boolean,
+            required: false,
+            default: false,
+        },
+        readonly: {
+            type: Boolean,
+            required: false,
+            default: false,
+        },
+        placeholder: {
+            type: String,
+            required: false,
+        },
+        color: {
+            type: String,
+            required: false,
+        },
+        styleObj: {
+            // This will be applied to the input element
+            type: Object,
+            required: false,
+            default: () => {},
+        },
+        datalist: {
+            // Display list of suggestions in autocomplete dialog
+            type: Array,
+            required: false,
+        },
+
+        // These last three are for handling special input cases only
+
+        optional: {
+            type: Boolean,
+            default: false,
+            required: false,
+        },
+        model_class: {
+            type: String,
+            required: false,
+        },
+        data: {
+            type: Object,  // ?
+            required: false,
+        },
+    },
+    data() {
+        if (
+            ["SelectTagParameter", "ColumnListParameter"].includes(this.model_class) ||
+            (this.options && this.data)
```

Comment:
I added in the logic for special `model_class` definitions here - refactored a bit to make it more concise. I added three more props to the component to make this work: `optional`, `model_class` and `data`. I assume you can't pass arbitrary props to a Vue component or this could be omitted?